torch/_dynamo/variables/tensor.py

```diff
@@ -591,6 +591,26 @@ def has_bool_key(v):
             )
             result = TorchVariable(torch.any, **options).call_function(tx, [result], {})
             return result.call_method(tx, "item", [], {})
+        elif name == "redistribute":
+            # rewrite non-primitive args/kwargs to be included in the on-the-fly prim function
+            # and rewrite args to have only proxyable args, then insert call_function
+            args_as_value = [x.as_python_constant() for x in args]
+
+            def redistribute_fn_with_prim_types(x):
+                return x.redistribute(*args_as_value)
+
+            # attach the same function name for better debugging
+            redistribute_fn_with_prim_types.__name__ = f"prim_{name}"
+
+            return wrap_fx_proxy(
+                tx=tx,
+                proxy=tx.output.create_proxy(
+                    "call_function",
+                    redistribute_fn_with_prim_types,
```

Comment:
Do you mind showing what a graph with this looks like? I am not certain you can just capture it like this and close over `args_as_value` soundly

Reply:
Sure here is the captured graph:

```python
    def forward(self, L_x_ : torch.Tensor):
        l_x_ = L_x_
        
        # File: /scratch/wanchaol/work/pytorch/test/distributed/_tensor/test_dtensor.py:685, code: dt = DTensor.from_local(x, mesh, [Shard(0)], run_check=False)
        prim_from_local = torch__dynamo_variables_torch_prim_from_local(l_x_, run_check = False);  l_x_ = None
        
        # File: /scratch/wanchaol/work/pytorch/test/distributed/_tensor/test_dtensor.py:686, code: return dt.redistribute(mesh, [Replicate()]).to_local() + 2
        prim_redistribute = torch__dynamo_variables_tensor_prim_redistribute(prim_from_local);  prim_from_local = None
        to_local = prim_redistribute.to_local();  prim_redistribute = None
        add = to_local + 2;  to_local = None
        return (add,)
```
since the `args` here are "device_mesh" and "placements" only (`redistribute` have a fixed signature), so the `args_as_value` would be safely encoded inside the `redistribute_fn_with_prim_types` and we insert a call_function node inside the graph.

Reply:
As long as this runs correctly, I am fine. I was skeptical, but its nice to see.